.github/workflows/pytest.yml

```diff
@@ -34,19 +36,22 @@ jobs:
 
       - uses: actions/cache@v2
         if: startsWith(runner.os, 'Windows')
+        id: cache_Windows
         with:
           path: ~\AppData\Local\pip\Cache
           key: ${{ runner.os }}-pip-${{ hashFiles('setup.cfg') }}
           restore-keys: |
             ${{ runner.os }}-pip-
 
       - name: pip installs
+        if: steps.cache_${{ runner.os }}.outputs.cache-hit != 'true'
         run: |
-          pip install pytest pytest-cov pytest-xdist setuptools_scm
-          pip install -e .
+          pip install -e .[test]
       - name: run pytest
         run: |
-          pytest --runslow -n2
+          pytest -n2 --dist loadscope --nbval --current-env tutorials/
+          pytest --runslow -n2 --cov-append tests/
```

Comment:
can we split this into two steps?
First one running the default tests and the second one running the notebooks? Should make it easier to spot where failures happen

I would also be fine with only running the notebooks if the normal tests succeed

Reply:
I split these up as demanded. I had to change the way the coverage is combined though